### PR TITLE
Remove most ARM NYI

### PR DESCRIPTION
**Pull request by BruceForstall**
Convert most existing ARM NYI either to asserts or remove
the code entirely.

A few NYI are left, including:
1. GT_BITCAST from 'int' to 'float'
2. initblk unrolling is unimplemented: https://github.com/dotnet/coreclr/issues/16349
3. SIMD-related NYI (SIMD is currently unimplemented)

### CODE REVIEW

**Comment by BruceForstall** [comment]
Fixes https://github.com/dotnet/coreclr/issues/13000

**Comment by BruceForstall** [comment]
@dotnet/jit-contrib

**Comment by RussKeldorph** [comment]
I can't figure out why the [Ubuntu x64 Checked job](https://ci.dot.net/job/dotnet_coreclr/job/master/job/x64_checked_ubuntu_innerloop_flow_prtest/2396/) failed.  Just a message `System.Private.CoreLib.dll does not exist.` with no obvious cause.  Retrying to see if it is repeatable.

@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test

File: src/jit/codegenarmarch.cpp [hunk 1]
```diff
@@ -334,22 +337,21 @@ void CodeGen::genCodeForTreeNode(GenTree* treeNode)
...
-            instGen_MemoryBarrier();
-            break;
-
         case GT_CMPXCHG:
-            NYI_ARM("GT_CMPXCHG");
-#ifdef _TARGET_ARM64_
             genCodeForCmpXchg(treeNode->AsCmpXchg());
-#endif
             break;
+#endif
```
Comment (sandreenko) [inline src/jit/codegenarmarch.cpp]: `// _TARGET_ARM64_ `

File: src/jit/codegenarmarch.cpp [hunk 2]
```diff
@@ -511,12 +513,12 @@ void CodeGen::genIntrinsic(GenTree* treeNode)
...
             break;
-#endif
+
         case CORINFO_INTRINSIC_Round:
-            NYI_ARM("genIntrinsic for round - not implemented yet");
             genConsumeOperands(treeNode->AsOp());
-            getEmitter()->emitInsBinary(INS_ROUND, emitActualTypeSize(treeNode), treeNode, srcNode);
+            getEmitter()->emitInsBinary(INS_frintn, emitActualTypeSize(treeNode), treeNode, srcNode);
             break;
+#endif
```
Comment (sandreenko) [inline src/jit/codegenarmarch.cpp]: `// _TARGET_ARM64_ `

File: src/jit/compiler.h [hunk 3]
```diff
@@ -10548,9 +10548,8 @@ const instruction INS_ADDC            = INS_adc;
 const instruction INS_SUBC            = INS_sbc;
 const instruction INS_NOT             = INS_mvn;
 
-const instruction INS_ABS   = INS_vabs;
-const instruction INS_ROUND = INS_invalid;
-const instruction INS_SQRT  = INS_vsqrt;
+const instruction INS_ABS  = INS_vabs;
+const instruction INS_SQRT = INS_vsqrt;
 
 #endif
```
Comment (sandreenko) [inline src/jit/compiler.h]: `// _TARGET_ARM_ `

File: src/jit/compiler.h [hunk 4]
```diff
@@ -10559,9 +10558,8 @@ const instruction INS_SQRT  = INS_vsqrt;
 const instruction INS_MULADD     = INS_madd;
 const instruction INS_BREAKPOINT = INS_bkpt;
 
-const instruction INS_ABS   = INS_fabs;
-const instruction INS_ROUND = INS_frintn;
-const instruction INS_SQRT  = INS_fsqrt;
+const instruction INS_ABS  = INS_fabs;
+const instruction INS_SQRT = INS_fsqrt;
 
 #endif
```
Comment (sandreenko) [inline src/jit/compiler.h]: `// _TARGET_ARM64_ `

**Comment by BruceForstall** [comment]
> do we plan to implement CORINFO_INTRINSIC_Interlocked* for arm32 in the future?

Don't know. https://github.com/dotnet/coreclr/issues/17075.

File: src/jit/lsraarm.cpp [hunk 5]
```diff
@@ -762,9 +762,8 @@ void LinearScan::BuildNode(GenTree* tree)
             _snprintf_s(message, _countof(message), _TRUNCATE, "NYI: Unimplemented node type %s",
                         GenTree::OpName(tree->OperGet()));
             NYIRAW(message);
-#else
-            NYI_ARM("BuildNode default case");
```
Comment (CarolEidt) [inline src/jit/lsraarm.cpp]: I would propose changing the `NYIRAW` to an assert, and then adding an `unreached()`. As this stands (unless I'm misreading), the default case will now fall through.

**Pull request merged**